src/v/debug_bundle/types.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "base/seastarx.h"
+#include "container/fragmented_vector.h"
+#include "model/metadata.h"
+#include "strings/string_switch.h"
+#include "utils/uuid.h"
+
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/sstring.hh>
+
+#include <absl/container/btree_set.h>
+#include <fmt/format.h>
+
+#include <chrono>
+#include <optional>
+#include <ostream>
+#include <variant>
+namespace debug_bundle {
+
+/// Special date strings used by journalctl
+enum class special_date { yesterday, today, now, tomorrow };
+
+constexpr std::string_view to_string_view(special_date d) {
+    switch (d) {
+    case special_date::yesterday:
+        return "yesterday";
+    case special_date::today:
+        return "today";
+    case special_date::now:
+        return "now";
+    case special_date::tomorrow:
+        return "tomorrow";
+    }
+}
+
+inline std::ostream& operator<<(std::ostream& o, const special_date& d) {
+    return o << to_string_view(d);
+}
+
+inline std::istream& operator>>(std::istream& i, special_date& d) {
+    ss::sstring s;
+    i >> s;
+    d = string_switch<special_date>(s)
+          .match(
+            to_string_view(special_date::yesterday), special_date::yesterday)
+          .match(to_string_view(special_date::today), special_date::today)
+          .match(to_string_view(special_date::now), special_date::now)
+          .match(
+            to_string_view(special_date::tomorrow), special_date::tomorrow);
+    return i;
+}
+
+/// Defines which clock the debug bundle will use
+using debug_bundle_clock = ss::lowres_system_clock;
+/// A timepoint
+using debug_bundle_timepoint = std::chrono::time_point<debug_bundle_clock>;
+/// When provided a time, may either be a time point or a special_date
+using time_variant = std::variant<debug_bundle_timepoint, special_date>;
+
+/// SCRAM credentials for authn
+struct scram_creds {
+    ss::sstring username;
+    ss::sstring password;
+    ss::sstring mechanism;
+};
```

Comment:
a: correct, they'll be passed on to the exec call

side-note: Update this, use acl_principal and mechanism type to avoid passing strings directly from Admin API (also will need to figure out something with the password)